Gems/Atom/Feature/Common/Assets/Materials/Types/DepthPass_WithPS.azsli

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#ifdef SHADOWS
+#include <Atom/Features/Shadow/Shadow.azsli>
+#endif
+
+struct VSInput
+{
+    float3 m_position : POSITION;
+    float2 m_uv0 : UV0;
+    float2 m_uv1 : UV1;
+
+    // only used for parallax depth calculation
+    float3 m_normal : NORMAL;
+    float4 m_tangent : TANGENT; 
+    float3 m_bitangent : BITANGENT; 
+
+#ifdef MULTILAYER
```

Comment:
From my understanding, you're creating DepthPass_WithPS.azsli to be a generic template that you can then customize by pre-including function definitions and #defines. Our current trend on Atom has been to avoid uber-shader type designs where possible, so that if someone is reading a material .azsl file and it's includes, it should as closely as possible contain only code that is used by the shader and nothing more. Having a specific material feature (MULTILAYER in this case) referenced inside this generic DepthPass_WithPS.azsli file goes against that trend (and you can see this expanding with things like #ifdef VERTEX_WIND_ANIMATION etc. Maybe that's what we need to do, switch trends in favor of a more uber-shader approach to fit the refactor you're aiming at, but if that's the direction you want to go then we should definitely have that discussion. Otherwise if the intention wasn't to go towards the uber-shader approach though, I suggest we avoid specific like MULTILAYER and ShouldHandleParallaxInDepthShaders in the generic shader and have more generic ways to add to the SRG, maybe along the lines of [ShouldHandleParallaxInDepthShaders](https://wiki.agscollab.com/display/ATOM/Generic+shader+code+injection+sites)

Reply:
https://github.com/o3de/o3de/pull/7437#discussion_r801076507